cmake/SetupProfiling.cmake

```diff
@@ -0,0 +1,21 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+option(KEEP_FRAME_POINTER, "Add keep frame pointer for profiling" OFF)
+
+add_library(Profiling::KeepFramePointer IMPORTED INTERFACE)
+
+if (KEEP_FRAME_POINTER)
+  set_property(
+    TARGET Profiling::KeepFramePointer
+    APPEND PROPERTY
+    INTERFACE_COMPILE_OPTIONS
+    $<$<COMPILE_LANGUAGE:CXX>:-fno-omit-frame-pointer>
```

Comment:
looking around a little bit, it seems like we might also want `-mno-omit-leaf-frame-pointer` -- what do you think?

Reply:
I've never received this advice for profiling code. Where did you see that advice being given? Are you referring to this LLVM bug: https://bugs.llvm.org/show_bug.cgi?id=9825 ?

Reply:
Yeah, that's what I found to suggest doing this, and it wasn't completely clear to me whether the fact the bug was closed would propagate to pre-2019 versions of clang.
My rough understanding from intel documentation and gcc documentation is that the `-fno-omit-frame-pointer` should imply the leaf option, it's just the clang bug that seemed to treat them differently.

Reply:
Alright, well no harm in adding the flag so we don't get different results using different compilers :)